engine/src/main/java/org/hibernate/validator/HibernateValidatorConfiguration.java

```diff
@@ -186,4 +187,19 @@
 	 * @since 5.2
 	 */
 	HibernateValidatorConfiguration timeProvider(TimeProvider timeProvider);
+
+	/**
+	 * Registers an expression factory for use when constraint violation messages use EL.
+	 * This allows applications to avoid issues with the method {@link ExpressionFactory#newInstance()},
+	 * which uses the TCCL in some implementations.
+	 * @param expressionFactory The expression factory for use.
+	 * @return  {@code this} following the chaining method pattern
+	 * @since 5.3
+     */
+	HibernateValidatorConfiguration expressionFactory(ExpressionFactory expressionFactory);
```

Comment:
@gunnarmorling , doesn't that still expose `ExpressionFactory` in just the way that @hferentschik dislikes? If the two of you agree on that, far be it from me to argue.